<a href="https://colab.research.google.com/github/Manversajjad/Pandas_exercise_LJMU_1/blob/main/StockPrice_AnalysisPandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Session 2
## Comprehension 3

Suppose as an analyst, your job is to analyze stock price data. You are provided the following lists:
*   **day_list** is a list of days you have data for
*   **apple_list**, **microsoft_list**, **meta_list**, **amazon_list**, and **tesla_list** are lists containing stock prices corresponding to the days in **day_list**

In [1]:
day_list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]

apple_list = [172.19, 175.08, 175.53, 172.19, 173.07, 169.8, 166.23, 164.51, 162.41, 161.62, 159.78, 159.69, 159.22, 170.33, 174.78]

microsoft_list = [314.27, 314.98, 318.27, 304.8, 310.2, 302.65, 303.33, 301.6, 296.03, 296.37, 288.49, 296.71, 299.84, 308.26, 310.98]

meta_list = [328.07, 334.37, 333.26, 326.48, 331.9, 318.15, 319.59, 316.56, 303.17, 308.71, 300.15, 294.63, 294.64, 301.71, 313.26]

amazon_list = [3229.72, 3307.24, 3304.14, 3224.28, 3242.76, 3178.35, 3125.98, 3033.35, 2852.86, 2890.88, 2799.72, 2777.45, 2792.75, 2879.56, 2991.47]

tesla_list = [1058.12, 1064.4, 1106.22, 1031.56, 1049.61, 1030.51, 995.65, 996.27, 943.9, 930, 918.4, 937.41, 829.1, 846.35, 936.72]

Execute the code below to create the DataFrame **stocks**, which will store the data.

In [2]:
# Importing the required library
import pandas as pd

# Defining the DataFrame
stocks = pd.DataFrame({'apple': apple_list, \
                       'microsoft': microsoft_list, \
                       'meta': meta_list, \
                       'amazon': amazon_list, \
                       'tesla': tesla_list}, \
                       index=day_list)


# Printing the DataFrame
stocks

apple  microsoft    meta   amazon    tesla
1   172.19     314.27  328.07  3229.72  1058.12
2   175.08     314.98  334.37  3307.24  1064.40
3   175.53     318.27  333.26  3304.14  1106.22
4   172.19     304.80  326.48  3224.28  1031.56
5   173.07     310.20  331.90  3242.76  1049.61
6   169.80     302.65  318.15  3178.35  1030.51
7   166.23     303.33  319.59  3125.98   995.65
8   164.51     301.60  316.56  3033.35   996.27
9   162.41     296.03  303.17  2852.86   943.90
10  161.62     296.37  308.71  2890.88   930.00
11  159.78     288.49  300.15  2799.72   918.40
12  159.69     296.71  294.63  2777.45   937.41
13  159.22     299.84  294.64  2792.75   829.10
14  170.33     308.26  301.71  2879.56   846.35
15  174.78     310.98  313.26  2991.47   936.72

Your team now wants to use your skills with DataFrames to derive insights from this data.
<br><br>

**Question 1**
<br>
Which of the following will return the stock prices for the last 5 days?

a. stocks.loc[-5:, :] \
b. stocks.head() \
c. stocks.iloc[11:15, :] \
d. stocks.tail()

In [3]:
# Write your code here
stocks.tail()

apple  microsoft    meta   amazon   tesla
11  159.78     288.49  300.15  2799.72  918.40
12  159.69     296.71  294.63  2777.45  937.41
13  159.22     299.84  294.64  2792.75  829.10
14  170.33     308.26  301.71  2879.56  846.35
15  174.78     310.98  313.26  2991.47  936.72

**Question 2**
<br>
Write code to find the average price for each stock.


In [4]:
# Write your code here. This method and the below defined method are the same to identify the Mean or average value.
stocks.describe()

apple   microsoft        meta       amazon        tesla
count   15.000000   15.000000   15.000000    15.000000    15.000000
mean   167.762000  304.452000  314.976667  3042.034000   978.281333
std      6.108603    8.338074   13.988310   198.552057    80.886219
min    159.220000  288.490000  294.630000  2777.450000   829.100000
25%    162.015000  298.275000  302.440000  2866.210000   933.360000
50%    169.800000  303.330000  316.560000  3033.350000   995.650000
75%    172.630000  310.590000  327.275000  3227.000000  1040.585000
max    175.530000  318.270000  334.370000  3307.240000  1106.220000

In [5]:
stocks.describe().loc['mean']

apple         167.762000
microsoft     304.452000
meta          314.976667
amazon       3042.034000
tesla         978.281333
Name: mean, dtype: float64

**Question 3**
<br>
What is the maximum stock price for Amazon over the given period?

a.   \\$3,307.24 \
b.   \\$3,042.03 \
c.   \\$3,178.35 \
d.   \\$3,424.28


In [8]:
# Write your code here
stocks.describe().loc['max']

apple         175.53
microsoft     318.27
meta          334.37
amazon       3307.24
tesla        1106.22
Name: max, dtype: float64

**Question 4**
<br>
Suppose a person purchases one stock of each company on Day 6. What would be the total value of their investment on that day?

a.   \\$4,970 \
b.   \\$4,999 \
c.   \\$5,003 \
d.   \$4,982

In [22]:
# Write your code here
total_purchase = stocks.sum(axis=1).loc[6]

print(f'Total purchase Amount investment on that day :- {total_purchase}$')


Total purchase Amount investment on that day :- 4999.46$


**Question 5**
<br>
Which of the following stocks has the highest positive correlation with Tesla?

a.   Apple \
b.   Microsoft \
c.   Meta \
d.   Amazon

In [18]:
# Write your code here. Answer is Meta, that Tesla has high correlation with Meta
highest_corr_stock = stocks.corr()
highest_corr_stock



apple  microsoft      meta    amazon     tesla
apple      1.000000   0.907103  0.826560  0.837784  0.657138
microsoft  0.907103   1.000000  0.764021  0.780525  0.591854
meta       0.826560   0.764021  1.000000  0.981416  0.911109
amazon     0.837784   0.780525  0.981416  1.000000  0.905781
tesla      0.657138   0.591854  0.911109  0.905781  1.000000